packages/web5/lib/src/vc/vc.dart

```diff
@@ -0,0 +1,153 @@
+  String issuanceDate;
+  // https://www.w3.org/TR/vc-data-model/#expiration
+  String? expirationDate;
+  // https://www.w3.org/TR/vc-data-model-2.0/#data-schemas
+  List<CredentialSchema>? credentialSchema;
+
+  VerifiableCredential._({
+    required this.context,
+    required this.type,
+    required this.issuer,
+    required this.subject,
+    required this.data,
+    required this.issuanceDate,
+    required this.id,
+    this.expirationDate,
+    this.credentialSchema = const [],
+  });
+
+  static VerifiableCredential create({
+    required String issuer,
+    required String subject,
+    required Map<String, dynamic> data,
+    List<String>? context,
+    List<String>? type,
+    String? id,
+    DateTime? issuanceDate,
+    DateTime? expirationDate,
+    List<CredentialSchema> credentialSchema = const [],
+  }) {
+    final uuid = Uuid();
+
+    context = context ?? [baseContext];
+    type = type ?? [baseType];
+    id = id ?? 'urn:vc:uuid:${uuid.v4()}';
+    issuanceDate = issuanceDate ?? DateTime.now();
+
+    return VerifiableCredential._(
+      context: context,
+      type: type,
+      issuer: issuer,
+      subject: subject,
+      data: data,
+      id: id,
+      issuanceDate: issuanceDate.toString(),
+      expirationDate: expirationDate?.toString(),
+      credentialSchema: credentialSchema,
+    );
+  }
+
+  Future<String> sign(
+    BearerDid bearerDid,
+  ) async {
+    final claims = JwtClaims(
+      iss: issuer,
+      jti: id,
+      sub: subject,
+    );
+
+    final issuanceDateTime = DateTime.parse(issuanceDate);
+    claims.nbf = issuanceDateTime.millisecondsSinceEpoch ~/ 1000;
+
+    if (expirationDate != null) {
+      final expirationDateTime = DateTime.parse(expirationDate!);
+      claims.exp = expirationDateTime.millisecondsSinceEpoch ~/ 1000;
+    }
+
+    claims.misc = {'vc': toJson()};
+
+    return await Jwt.sign(did: bearerDid, payload: claims);
+  }
+
+  factory VerifiableCredential.fromJson(Map<String, dynamic> json) {
+    final credentialSubject = json['credentialSubject'] as Map<String, dynamic>;
+    final id = credentialSubject.remove('id');
+    final credentialSchema = (json['credentialSchema'] as List<dynamic>)
+        .map((e) => CredentialSchema(id: e['id'], type: e['type'])).toList();
+    final context = (json['@context'] as List<dynamic>).cast<String>();
+    final type = (json['type'] as List<dynamic>).cast<String>();
+
+    return VerifiableCredential._(
+      issuer: json['issuer'],
+      subject: json['subject'],
+      data: credentialSubject,
+      id: id,
+      context: context,
+      type: type,
+      issuanceDate: json['issuanceDate'],
+      expirationDate: json['expirationDate'],
+      credentialSchema: credentialSchema,
+    );
+  }
+
+  Map<String, dynamic> toJson() {
+    return {
+      '@context': context,
+      'subject': subject,
+      'type': type,
+      'issuer': issuer,
+      'credentialSubject': {
+        'id': 'id',
```

Comment:
i'm guessing this is a placeholder? we need some way to guarantee that it's set per [web5 spec](https://github.com/TBD54566975/web5-spec/blob/main/spec/vc.md#verifiable-credential-data-model)

Reply:
Ope, that was definitely a mistake. Just double checked the spec and should be fixed now